mopidy_musicbox_webclient/static/js/controls.js

```diff
@@ -207,37 +207,68 @@ function removeTrack() {
 }
 
 function clearQueue() {
-    mopidy.playback.stop();
-    resetSong();
-    mopidy.tracklist.clear();
-    resetSong();
+    mopidy.playback.stop().then(
+        mopidy.tracklist.clear().then(
+            resetSong()
+        )
+    );
     return false;
 }
 
-function saveQueue() {
+function savePressed(key) {
+    if (key == 13) {
+        saveQueue();
+        return false;
+    }
+    return true;
+}
+
+function showSavePopup(){
     mopidy.tracklist.getTracks().then(function(tracks) {
         if (tracks.length > 0) {
-            var plname = window.prompt("Playlist name:", "").trim();
-            if (plname != null && plname != "") {
-                mopidy.playlists.filter({"name": plname}).then(function(existing) {
-                    var exists = false;
-                    for (var i = 0; i < existing.length; i++) {
-                        exists = exists || existing[i].uri.indexOf("m3u:") == 0 || existing[i].uri.indexOf("local:") == 0;
-                    }
-                    if (!exists || window.confirm("Overwrite existing playlist \"" + plname + "\"?")) {
-                        mopidy.playlists.create({'name': plname, 'uri_scheme': "local"}).then(function(playlist) {
-                             playlist.tracks = tracks;
-                             mopidy.playlists.save({'playlist': playlist}).then();
-                             getPlaylists();
-                        });
-                    }
-                });
-            }
+            $('#saveinput').val('');
+            $('#popupSave').popup('open');
+        }
+    });
+
+}
+
+function saveQueue() {
+    mopidy.tracklist.getTracks().then(function(tracks) {
+        var plname = $('#saveinput').val().trim();
+        if (plname != null && plname != "") {
+            mopidy.playlists.filter({"name": plname}).then(function(existing) {
+                var exists = false;
+                for (var i = 0; i < existing.length; i++) {
+                    exists = exists || existing[i].uri.indexOf("m3u:") == 0 || existing[i].uri.indexOf("local:") == 0;
+                }
+                if (exists) {
+                    $('#popupSave').popup('close');
+                    $('#popupOverwrite').popup('open');
+                    $('#overwriteConfirmBtn').click(function() {
+                        initSave(plname, tracks);
+                    });
+                } else {
+                    initSave(plname, tracks);
+                }
+            });
         }
     });
     return false;
 }
 
+function initSave(playlistName, tracks) {
+    $('#popupOverwrite').popup('close');
+    $('#popupSave').popup('close');
+    $('#saveinput').val('');
+    toast('Saving...');
+    mopidy.playlists.create({'name': playlistName, 'uri_scheme': "local"}).then(function(playlist) {
```

Comment:
Again, not your doing, but using this the local uri scheme isn't right (anymore).